presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java

```diff
@@ -78,8 +85,7 @@ public int link(int from, int to)
             // make sure that from value is the smaller one
             if (comparator.compare(from, to) > 0) {
                 // _from_ is larger so, just add to current chain _to_
-                List<Integer> links = positionLinks.computeIfAbsent(to, key -> new IntArrayList());
-                links.add(from);
+                positionLinks.computeIfAbsent(to, NEW_INT_LIST).add(from);
```

Comment:
If we want to avoid get/put additional hash/probe, would `compute` work better?
```
positionLinks.compute(to, (key, value) -> (value == null ? NEW_INT_LIST : value).add(item))
```